airflow/contrib/operators/kubernetes_operator.py

```diff
@@ -352,6 +364,9 @@ def create_job_yaml(self, context):
                 'secret': {'secretName': self.service_account_secret_name},
             })
 
+        # add in all the volumes given by the user, assuming they don't conflict with the secrets volumes we created
+        volumes += [v for v in self.volumes if v['name'] not in {v['name'] for v in volumes}]
```

Comment:
Ironic that I'm suggesting a one-liner on line 146; I find this doubly-listed list/dict comprehension hard to understand. I would find it more readable if it were expanded into a multi-line version with boring for loops, but I'll leave it to you to decide

Reply:
done